catalog/core/catalog-core-subscriptionstore/src/main/java/org/codice/ddf/catalog/subscriptionstore/SubscriptionCache.java

```diff
@@ -0,0 +1,234 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.subscriptionstore;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang.Validate.notEmpty;
+import static org.apache.commons.lang.Validate.notNull;
+
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import javax.annotation.Nullable;
+
+import org.codice.ddf.catalog.subscriptionstore.internal.SubscriptionContainer;
+import org.codice.ddf.catalog.subscriptionstore.internal.SubscriptionFactory;
+import org.codice.ddf.catalog.subscriptionstore.internal.SubscriptionStoreException;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceRegistration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.event.Subscription;
+
+/**
+ * Container providing centralized, cached access to all subscriptions that can be registered with
+ * the system from any endpoint or subscription provider.
+ */
+public class SubscriptionCache implements SubscriptionContainer {
```

Comment:
I want to do more verification. From initial testing it appeared to work fine, and that makes sense since the endpoints are still Subscribers (have the delete method) and redundant deletes are no-ops. 

A valid subscription type is needed for any invocation of the internal API, so the behavior is still similar to the managed-service factory approach. Other endpoints can only interact with **their** subscriptions.